### PR TITLE
support libvirt DNS/DHCP provider

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: spec/classes/foreman_proxy__config__spec.rb [hunk 1]
```diff
@@ -68,12 +68,21 @@
         end
 
         it 'should create configuration files' do
-          [ "#{etc_dir}/foreman-proxy/settings.yml", "#{etc_dir}/foreman-proxy/settings.d/tftp.yml", "#{etc_dir}/foreman-proxy/settings.d/dns.yml",
-            "#{etc_dir}/foreman-proxy/settings.d/dns_nsupdate.yml", "#{etc_dir}/foreman-proxy/settings.d/dns_nsupdate_gss.yml",
-            "#{etc_dir}/foreman-proxy/settings.d/dhcp.yml", "#{etc_dir}/foreman-proxy/settings.d/dhcp_isc.yml",
-            "#{etc_dir}/foreman-proxy/settings.d/puppetca.yml", "#{etc_dir}/foreman-proxy/settings.d/puppet.yml",
-            "#{etc_dir}/foreman-proxy/settings.d/bmc.yml", "#{etc_dir}/foreman-proxy/settings.d/realm.yml",
-            "#{etc_dir}/foreman-proxy/settings.d/templates.yml", "#{etc_dir}/foreman-proxy/settings.d/logs.yml"].each do |cfile|
+          [ "#{etc_dir}/foreman-proxy/settings.yml",
```
Comment (ekohl) [inline spec/classes/foreman_proxy__config__spec.rb]: Could be simplified to just ```[ 'settings.yml', 'settings.d/bmc.yml', ...].each do |cfile| should contain_file("#{etc_dir}/foreman-proxy/#{cfile}) end ```

File: README.md [hunk 2]
```diff
@@ -28,6 +28,10 @@ configuration layout changed significantly.
 To configure older versions of the Smart Proxy (1.5 to 1.10), use version 2.x
 of this module.
 
+### 1.11 compatibility notes
```
Comment (ekohl) [inline README.md]: This also works on 1.10 I think.

**Comment by ekohl** [comment]
Other than small tweaks :+1:

File: README.md (hunk 2 above)
Comment (mmoll) [inline README.md]: other things were removed, so 3.x won't work for 1.10

**Comment by ekohl** [comment]
:+1:

**Comment by domcleal** [comment]
Merged, thanks @mmoll!

**Pull request merged**